app/controllers/stories_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class StoriesController < ApplicationController
+  before_action :update_top_stories, only: :index
+
+  def index
+    @stories = Story.where(hacker_news_id: top_stories_ids).includes(:users)
+  end
+
+  private
+
+  def top_stories
+    @top_stories ||= Story.where(hacker_news_id: top_stories_ids)
+  end
+
+  def top_stories_ids
```

Comment:
Render the top 50 stories, I could add pagination to be able to page through all 500 if required.